In [17]:
import csv
import datetime

In [18]:
data = list(csv.reader(open('guns.csv','r')))

In [19]:
headers = data[0]
data = data[1:]

In [20]:
print(headers)
print(data[0:5])

['', 'year', 'month', 'intent', 'police', 'sex', 'age', 'race', 'hispanic', 'place', 'education']
[['1', '2012', '01', 'Suicide', '0', 'M', '34', 'Asian/Pacific Islander', '100', 'Home', '4'], ['2', '2012', '01', 'Suicide', '0', 'F', '21', 'White', '100', 'Street', '3'], ['3', '2012', '01', 'Suicide', '0', 'M', '60', 'White', '100', 'Other specified', '4'], ['4', '2012', '02', 'Suicide', '0', 'M', '64', 'White', '100', 'Home', '4'], ['5', '2012', '02', 'Suicide', '0', 'M', '31', 'White', '100', 'Other specified', '2']]


In [21]:
years = [x[1] for x in data]

year_counts = {}

for i in years:
    if i in year_counts:
        year_counts[i] += 1
    else:
        year_counts[i] = 1

print(year_counts)

{'2013': 33636, '2014': 33599, '2012': 33563}


In [22]:
dates = [datetime.datetime(year=int(x[1]), month=int(x[2]), day=1) for x in data]

In [23]:
print(dates[0:5])

[datetime.datetime(2012, 1, 1, 0, 0), datetime.datetime(2012, 1, 1, 0, 0), datetime.datetime(2012, 1, 1, 0, 0), datetime.datetime(2012, 2, 1, 0, 0), datetime.datetime(2012, 2, 1, 0, 0)]


In [24]:
date_counts = {}

for i in dates:
    if i in date_counts:
        date_counts[i] += 1
    else:
        date_counts[i] = 1

print(date_counts)

{datetime.datetime(2012, 9, 1, 0, 0): 2852, datetime.datetime(2014, 4, 1, 0, 0): 2862, datetime.datetime(2014, 2, 1, 0, 0): 2361, datetime.datetime(2012, 5, 1, 0, 0): 2999, datetime.datetime(2012, 7, 1, 0, 0): 3026, datetime.datetime(2014, 1, 1, 0, 0): 2651, datetime.datetime(2012, 2, 1, 0, 0): 2357, datetime.datetime(2014, 10, 1, 0, 0): 2865, datetime.datetime(2014, 8, 1, 0, 0): 2970, datetime.datetime(2013, 1, 1, 0, 0): 2864, datetime.datetime(2013, 8, 1, 0, 0): 2859, datetime.datetime(2013, 12, 1, 0, 0): 2765, datetime.datetime(2012, 12, 1, 0, 0): 2791, datetime.datetime(2013, 10, 1, 0, 0): 2808, datetime.datetime(2013, 9, 1, 0, 0): 2742, datetime.datetime(2014, 3, 1, 0, 0): 2684, datetime.datetime(2013, 6, 1, 0, 0): 2920, datetime.datetime(2014, 9, 1, 0, 0): 2914, datetime.datetime(2013, 7, 1, 0, 0): 3079, datetime.datetime(2014, 11, 1, 0, 0): 2756, datetime.datetime(2013, 4, 1, 0, 0): 2798, datetime.datetime(2014, 12, 1, 0, 0): 2857, datetime.datetime(2012, 8, 1, 0, 0): 2954, date

In [25]:
sex_counts = {} # i[5]
race_counts = {} # i[7]

for i in data:
    if i[5] in sex_counts:
        sex_counts[i[5]] += 1
    else:
        sex_counts[i[5]] = 1
    
    if i[7] in race_counts:
        race_counts[i[7]] += 1
    else:
        race_counts[i[7]] = 1

print(race_counts)
print(sex_counts)

{'White': 66237, 'Hispanic': 9022, 'Asian/Pacific Islander': 1326, 'Black': 23296, 'Native American/Native Alaskan': 917}
{'F': 14449, 'M': 86349}


### Insights
While year-over-year gun deaths seemed to be consistent, it appears that gun deaths spike each year during the warmer months (May - Sept). The majority of people involved in gun deaths over the 3 years were White, with the minority being Native American/Native Alaskan. Males were involved in almost 6 times more gun deaths than females.

### Further examination
It would be beneficial to understand the trends in gun deaths based on BOTH race and gender together instead of separately.

In [26]:
census = list(csv.reader(open('census.csv','r')))
census

[['Id',
  'Year',
  'Id',
  'Sex',
  'Id',
  'Hispanic Origin',
  'Id',
  'Id2',
  'Geography',
  'Total',
  'Race Alone - White',
  'Race Alone - Hispanic',
  'Race Alone - Black or African American',
  'Race Alone - American Indian and Alaska Native',
  'Race Alone - Asian',
  'Race Alone - Native Hawaiian and Other Pacific Islander',
  'Two or More Races'],
 ['cen42010',
  'April 1, 2010 Census',
  'totsex',
  'Both Sexes',
  'tothisp',
  'Total',
  '0100000US',
  '',
  'United States',
  '308745538',
  '197318956',
  '44618105',
  '40250635',
  '3739506',
  '15159516',
  '674625',
  '6984195']]

In [30]:
mapping = {}

for i in race_counts:
    if i == 'Asian/Pacific Islander':
        mapping[i] = int(census[1][14]) + int(census[1][15])
    elif i == 'Black':
        mapping[i] = int(census[1][12])
    elif i == 'Hispanic':
        mapping[i] = int(census[1][11])
    elif i == 'Native American/Native Alaskan':
        mapping[i] = int(census[1][13])
    elif i == 'White':
        mapping[i] = int(census[1][10])

print(mapping)

{'White': 197318956, 'Black': 40250635, 'Asian/Pacific Islander': 15834141, 'Hispanic': 44618105, 'Native American/Native Alaskan': 3739506}


In [31]:
race_per_hundredk = {}

for i in race_counts:
    race_per_hundredk[i] = (float(race_counts[i]) / float(mapping[i])) * 100000
    
print(race_per_hundredk)

{'White': 33.56849303419181, 'Black': 57.8773477735196, 'Asian/Pacific Islander': 8.374309664161762, 'Hispanic': 20.220491210910907, 'Native American/Native Alaskan': 24.521955573811088}


In [32]:
intents = [x[3] for x in data]
races = [x[7] for x in data]
homicide_race_per_hundredk = {}

for i, race in enumerate(races):
    if intents[i] == 'Homicide':
        if race not in homicide_race_per_hundredk:
            homicide_race_per_hundredk[race] = 1
        else:
            homicide_race_per_hundredk[race] += 1

for i in race_counts:
    homicide_race_per_hundredk[i] = (float(homicide_race_per_hundredk[i]) / float(mapping[i])) * 100000

print(homicide_race_per_hundredk)

{'White': 4.6356417981453335, 'Hispanic': 12.627161104219914, 'Asian/Pacific Islander': 3.530346230970155, 'Black': 48.471284987180944, 'Native American/Native Alaskan': 8.717729026240365}


### Findings
Gun-related deaths are prevalent among individuals in the Black racial category, compared to other racial categories - with approximately 57.9 per 100,000 deaths occurring between 2012 and 2014. The next highest rate of gun-related deaths per 100,000 involves individuals in the White racial category (at approximately 33.6 deaths per 100,000).

When honing in on homicide-related gun deaths, the Black racial category still tops the list at approximately 48.5 deaths per 100,000. Surprisingly, the next highest homicide-related gun deaths belongs to the Hispanic racial category at 12.6 deaths per 100,000 (even though that category is fourth in overall gun-related deaths per 100,000).

### Next Steps
The following are a few ideas that are a work in progress on this analysis.
- Incorporate individuals of mixed race into the analysis to identify patterns involving mixed races
- Identify prevalent racial categories involved in other types of gun-deaths (e.g. are particular racial categories more prone to committing suicide?)
- Understand if there are seasonal patterns in gun deaths by racial category for particular intents (e.g. are suicides highest among particular racial categories in the winter?)
- Explore homicide rates by gender.
- Find out if gun death rates correlate to location and education.